scripts/self-hosted-kind/setup-host.yaml

```diff
@@ -0,0 +1,132 @@
+# MIT License
+#
+# Copyright (c) 2021 Mert Bora Alper and EASE lab
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+# Execute this playbook using 
+#   ansible-playbook -u YOUR_SSH_USERNAME -i REMOTE_HOSTNAME, setup-host.yaml
+
+---
+- name: Setup Knative-on-KinD Host
+  hosts: all
+  tasks:
+    - name: Add git PPA
+      become: yes
+      ansible.builtin.apt_repository:
+        repo: ppa:git-core/ppa
+
+    - name: Upgrade packages
+      become: yes
+      apt:
+        update_cache: yes
+        upgrade: safe
+
+    - name: Install required packages
+      become: yes
+      apt:
+        name:
+          - docker.io
+          - git-lfs
+          - mosh
+
+    - name: Add user "{{ ansible_user }}" to group docker
+      become: yes
+      ansible.builtin.user:
+        name: "{{ ansible_user }}"
+        groups: docker
+        append: yes
+
+    - name: Reset SSH connection to activate changes
+      meta: reset_connection
+
+    - name: Create /mydata/docker directory
+      become: yes
+      ansible.builtin.file:
+        path: /mydata/docker
+        state: directory
+        mode: '0755'
+
+    - name: Set docker storage to /mydata
+      become: yes
+      copy:
+        dest: /etc/docker/daemon.json
+        content: |
+          {
+            "data-root": "/mydata/docker"
+          }
+
+    - name: Restart Docker daemon to activate changes
+      become: yes
+      ansible.builtin.systemd:
+        state: restarted
+        name: docker
+
```

Comment:
Those are Cloudlab-specific, I shall remove them if we are going to use our own machines (provided by the University).

Reply:
please test them there. Shyam can deploy them on one of our nodes later